src/model/CoilHeatingElectricMultiStage.hpp

```diff
@@ -0,0 +1,111 @@
+*
+*  (3) Neither the name of the copyright holder nor the names of any contributors may be used to endorse or promote products
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_COILHEATINGELECTRICMULTISTAGE_HPP
+#define MODEL_COILHEATINGELECTRICMULTISTAGE_HPP
+
+#include "ModelAPI.hpp"
+#include "StraightComponent.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+  class Schedule;
+  class CoilHeatingElectricMultiStageStageData;
+
+  namespace detail {
+
+    class CoilHeatingElectricMultiStage_Impl;
+
+  }  // namespace detail
+
+  /** CoilHeatingElectricMultiStage is a StraightComponent that wraps the OpenStudio IDD object 'OS:Coil:Heating:Electric:MultiStage'.
+ * Performance data for each stage is contained in a separate ModelObject called OS:Coil:Heating:Electric:MultiStage:StageData **/
+  class MODEL_API CoilHeatingElectricMultiStage : public StraightComponent
+  {
+   public:
+    /** @name Constructors and Destructors */
+    //@{
+
+    explicit CoilHeatingElectricMultiStage(const Model& model);
+
+    virtual ~CoilHeatingElectricMultiStage() {}
+
+    //@}
+
+    static IddObjectType iddObjectType();
+
+    /** @name Getters */
+    //@{
+
+    boost::optional<Schedule> availabilitySchedule() const;
+
+    //@}
+    /** @name Setters */
+    //@{
+
+    bool setAvailabilitySchedule(Schedule& schedule);
+
+    void resetAvailabilitySchedule();
+
+    //@}
+    /** @name Other */
+    //@{
+
+    std::vector<CoilHeatingElectricMultiStageStageData> stages() const;
+
+    void addStage(CoilHeatingElectricMultiStageStageData& stage);
```

Comment:
I think this API is a bit too incomplete. At least provide

```c++
    unsigned numberOfStages() const;
    boost::optional<unsigned> stageIndex(const CoilHeatingElectricMultiStageStageData& stage) const;
    bool setStages(const std::vector<CoilHeatingElectricMultiStageStageData>& stages);
    void removeAllStages();
```

And maybe

```c++
    bool addStage(const CoilHeatingElectricMultiStageStageData& stage, unsigned index);
    bool setStageIndex(const CoilHeatingElectricMultiStageStageData& stage, unsigned index);
```


cf https://github.com/NREL/OpenStudio/blob/866817ccce3759a40601941d7919a54a94b778a2/src/model/CoilCoolingDXMultiSpeed.hpp#L133-L182